.github/workflows/dotnet.yml

```diff
@@ -22,3 +22,54 @@ jobs:
       run: |
         cd src
         & ".\nugetDevBuild.bat"
+
+  push:
+    needs: build
+    runs-on: ubuntu-22.04
+    env:
+      BASE_VERSION: 4.20.0
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 8.0.x
+
+    - name: Add dependencies
+      run: |
+        dotnet new classlib --name Nethereum --framework netstandard2.0
+        cd Nethereum
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.ABI
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.Accounts
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.Contracts
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.Hex
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.KeyStore
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.RLP
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.RPC
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.Signer
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.StandardTokenEIP20
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.Util
+        dotnet add package --version ${{env.BASE_VERSION}} Nethereum.Web3
+        rm Class1.cs
+
+    - name: Pack
+      run: |
+        cd Nethereum
+
+        VERSION="$(dotnet fsi ../nugetPreRelease.fsx $BASE_VERSION)"
+
+        dotnet pack \
+          --configuration Release \
+          --output ../dist \
+          --property:IncludeBuildOutput=false \
+          --property:IncludeContentInPack=false \
+          --property:TargetFrameworks=netstandard2.0 \
+          --property:Version=$VERSION
+    
+    - name: Push
```

Comment:
@webwarrior-ws please add an `if` here that has three conditions:
1. checks secrets.NUGET_API_KEY is not empty
2. checks github.event_name == 'push'
3. checks github.ref == 'refs/heads/upstream'

Reply:
Done